numba/unicode.py

```diff
@@ -1388,6 +1389,39 @@ def impl(s):
     return impl
 
 
+# https://github.com/python/cpython/blob/1d4b6ba19466aba0eb91c4ba01ba509acf18c723/Objects/unicodeobject.c#L11975-L12006    # noqa: E501
+@overload_method(types.UnicodeType, 'isalnum')
+def unicode_isalnum(data):
+    """Implements UnicodeType.isalnum()"""
+
+    def impl(data):
+        length = len(data)
+        if length == 0:
+            return False
+
+        if length == 1:
+            code_point = _get_code_point(data, 0)
```

Comment:
there's no ascii fast path in this branch?

Reply:
Added ascii fast path in this branch